components/style/traversal.rs

```diff
@@ -232,6 +234,32 @@ pub fn recalc_style_at<'a, N, C>(context: &'a C,
                 node.set_restyle_damage(damage);
             }
         }
+    } else {
+        // Finish any expired transitions, without performing the cascade.
+        let node_opaque = node.opaque();
+        let context = context.shared_context();
+
+        let had_animations_to_expire;
+        {
+            let mut data = node.mutate_data().unwrap();
+            let mut style = data.style.as_mut().unwrap();
+            let all_expired_animations = context.expired_animations.read().unwrap();
+            let animations_to_expire = all_expired_animations.get(&node_opaque);
+            had_animations_to_expire = animations_to_expire.is_some();
+            if let Some(ref animations) = animations_to_expire {
+                for animation in *animations {
+                    // NB: Expiring a keyframes animation is the same as not
+                    // applying the keyframes style to it, so we're safe.
```

Comment:
This is not quite true, could you make a note about supporting [animation fill mode](https://www.w3.org/TR/css3-animations/#animation-fill-mode)?